crates/astria-conductor/src/client_provider.rs

```diff
@@ -0,0 +1,134 @@
+use async_trait::async_trait;
+use color_eyre::eyre::{
+    self,
+    WrapErr as _,
+};
+use deadpool::managed::{
+    self,
+    Pool,
+};
+use sequencer_client::WebSocketClient;
+use tokio::{
+    select,
+    sync::{
+        mpsc,
+        oneshot,
+    },
+    task::JoinHandle,
+};
+use tracing::warn;
+
+type ClientRx = mpsc::UnboundedReceiver<oneshot::Sender<Result<WebSocketClient, Error>>>;
+type ClientTx = mpsc::UnboundedSender<oneshot::Sender<Result<WebSocketClient, Error>>>;
+
+pub(super) async fn start_pool(url: &str) -> eyre::Result<Pool<ClientProvider>> {
```

Comment:
what's the purpose of using a pool over just having each component have their own `ClientProvider`?

Reply:
You mean having one open websocket connection per component?

The object pool with the provider solves two things:

1. You only have one websocket connection to the sequencer (that's just good practice and the reason to use websocket in the first place).
2. You ensure that there is a cap on the maximum number of requests to sequencer alive at the same time.

I hope this answers the question?

Reply:
This pool also solves for reconnect as well right? If each component had their own websocket connection (bad), they would also need their own reconnect logic.

Reply:
Not the pool but the ClientProvider.

But yes, if each actor got its own provider, we'd have one independent task per actor each, with each attempting to fix their own websocket connection at probably the same time.

Reply:
> You mean having one open websocket connection per component?

yeah, that answers my question :) thanks

Reply:
Yeah, you could have an `Arc<ConnectionProvider>` and then `get` connections from it if failed. But that's basically turning into a pool.